framework/Optimizers/stepManipulators/ConjugateGradient.py

```diff
@@ -15,7 +15,7 @@
   Step size manipulations based on gradient history
 
   Created 2020-01
-  @author: zhoujia
+  @author: zhoujia, alfoa
```

Comment:
This is a more general question; what is the process for updating authorship via module documentation? It seems to me it's rather arbitrary throughout the code, and redundant given the git history. I certainly have not updated this authorship to all files I've made adjustments in.

Reply:
If there is a contribution that changes the functionality of the peace of code (aka in this case to make it work as it was supposed to work from the beginning), the authorship should be updated.  If it is a recast of the software (like moving files from modules to standalone files, extract base classes and create new modules containing them, etc,) the original author must be preserved. If the recast of the software also add functionalities, the new "designer" or "implementer" should be added in the author list. The author list is also a way to define "responsable" developer of a certain module.

Are you not okay with me updating this file with my name?  

Anyhow, you made me think that we should probably remove the authors from all the modules.

Reply:
I've never been clear on the criteria for these; I don't personally mind whose name is there as long as it's largely aesthetic.

If we have a functional use for these, then I can try and find all the modules I've reworked and add my name to them (I suspect others will want to as well); otherwise, like I say, I'm not really bothered by whose name goes there. Not requesting any changes.

Reply:
No, clearly it determines an "acquiring of ownership" that is misleading. When in reality it should be more "who should I bother in case of something goes south" 

I will propose to the team to remove all the names to avoid it.

Reply:
I do like the idea of "primary maintainer" for new developers, although that fosters a divide-and-conquer approach to the code that we've tried to avoid previously.

In practice, though, we all figure out after a while who to go to about particular modules, so having this in weak documentation like this may be a good idea. Great idea to discuss it; I'll resolve this for the PR's sake.

Reply:
I've never been clear on the criteria for these either...Not request any changes, but if we want to build a clear git history, maybe we could do it like the tests file, add the revision comment?

Reply:
I wrote the criteria in the previous comment. If we are not clear about something, we can always ask.